src/backend/distributed/shardsplit/shardsplit_decoder.c

```diff
@@ -62,6 +75,68 @@ _PG_output_plugin_init(OutputPluginCallbacks *cb)
 	/* actual pgoutput callback will be called with the appropriate destination shard */
 	pgoutputChangeCB = cb->change_cb;
 	cb->change_cb = split_change_cb;
+	cb->filter_by_origin_cb = replication_origin_filter_cb;
+}
+
+
+/*
+ * replication_origin_filter_cb call back function filters out publication of changes
+ * originated from any other node other than the current node. This is
+ * identified by the "origin_id" of the changes. The origin_id is set to
+ * a non-zero value in the origin node as part of WAL replication.
+ */
+static bool
+replication_origin_filter_cb(LogicalDecodingContext *ctx, RepOriginId origin_id)
+{
+	if (origin_id != InvalidRepOriginId)
+	{
+		return true;
+	}
+	return false;
+}
+
+
+/*
+ * PublishChangesIfCdcSlot checks if the current slot is a CDC slot. If so, it publishes
+ * the changes as the change for the distributed table instead of shard.
+ * If not, it returns false. It also skips the Citus metadata tables.
+ */
+static bool
+PublishChangesIfCdcSlot(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
+						Relation relation, ReorderBufferChange *change)
+{
+	char *replicationSlotName = ctx->slot->data.name.data;
+
+	/* Check if the replication slot is CITUS_CDC_SLOT*/
+	if (replicationSlotName != NULL && strcmp(replicationSlotName, CITUS_CDC_SLOT_NAME) ==
```

Comment:
I'm kind of wondering whether we need to use a specific slot name, e.g. what if we want multiple CDC streams?

We're already using an output plug-in that is separate from pgouput, so the only thing this could conflict with is shard splits. Could we perhaps follow this code path whenever the slot does not belong to a split?

Reply:
+1 for that.  When we do individual table publications (e.g., not PUBLISH ALL), we'd need this anyway?

Also, related, do we currently need to manually create `citus_cdc_slot`? I guess at a minimum we should do that after user does a CREATE PUBLICATION .. ALL TABLES?

Reply:
> If users have created their own specific replication slot for some other reason than CDC, then we don't want to apply these changes to those slots.. So I chose to have a particular replication slot name. Is that ok?

I see this more as a behaviour of the citus output plug-in, which the user would choose along with the replication slot name.

Reply:
Ok, I have added the condition to check for the prefix "citus_shard_" to recognize the slots created by the plugin itself for shard moves/splits. Any other name will be treated as a slot for CDC.